core/src/main/scala/cats/data/Chain.scala

```diff
@@ -331,7 +331,7 @@ sealed abstract class Chain[+A] {
     case Singleton(a) => Singleton((a, 0))
     case Append(left, right) =>
       val leftSize = left.length.toInt
-      Append(left.zipWithIndex, right.zipWithIndex.map { case (a, i) => (a, leftSize + i) })
+      Wrap(left.toList.zipWithIndex ++ right.toList.zipWithIndex.map { case (a, i) => (a, leftSize + i) })
```

Comment:
Any reason not to use a `.toVector` or something else that `++`'s better than `List` here?